packages/react-components/react-rating-preview/src/components/RatingItem/RatingItem.test.tsx

```diff
@@ -1,10 +1,148 @@
+      step: 1,
+      size: 'medium',
+    };
+    const result = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={4} id="test" />
+      </RatingItemProvider>,
+    );
+    const items = result.container.querySelector('#test');
+    const child = items?.querySelector('.fui-RatingItem__unselectedFilledIcon');
+    expect(child).toBeTruthy();
+  });
+  it('renders the unselectedOutlineIcon slot for unselected items when not interactive', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: false,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const result = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={4} id="test" />
+      </RatingItemProvider>,
+    );
+    const items = result.container.querySelector('#test');
+    const child = items?.querySelector('.fui-RatingItem__unselectedOutlineIcon');
+    expect(child).toBeTruthy();
+  });
+  it('renders the unselectedOutlineIcon slot for unselected items when interactive', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: true,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const result = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={4} id="test" />
+      </RatingItemProvider>,
+    );
+    const items = result.container.querySelector('#test');
+    const child = items?.querySelector('.fui-RatingItem__unselectedOutlineIcon');
+    expect(child).toBeTruthy();
+  });
+  it('renders the selectedFilledIcon slot for selected items when interactive', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: true,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const result = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2} id="test" />
+      </RatingItemProvider>,
+    );
+    const items = result.container.querySelector('#test');
+    const child = items?.querySelector('.fui-RatingItem__selectedIcon');
+    expect(child).toBeTruthy();
+  });
+  it('does not render input elements when interactive is false', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: false,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const result = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2} id="test" />
+      </RatingItemProvider>,
+    );
+    const items = result.container.querySelector('#test');
+    const child = items?.querySelector('input');
```

Comment:
Again, prefer `getByRole('input')` / `getAllByRole('input')` instead of querySelector.